controllers/snapshot/snapshot_adapter.go

```diff
@@ -421,6 +498,41 @@ func (a *Adapter) updateExistingSnapshotEnvironmentBindingWithSnapshot(snapshotE
 	if err != nil {
 		return nil, err
 	}
-
 	return snapshotEnvironmentBinding, nil
 }
+
+//createCopyOfExistingEnvironment uses existing env as input, specifies namespace where the environment is situated,
+//integrationTestScenario contains information about existing environment
+//snapshot is mainly used for adding labels
+//returns copy of already existing environment with updated envVars
+func (a *Adapter) createCopyOfExistingEnvironment(existingEnvironment *applicationapiv1alpha1.Environment, namespace string, integrationTestScenario *v1alpha1.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot) (*applicationapiv1alpha1.Environment, error) {
+	environment := gitops.NewCopyOfExistingEnvironment(existingEnvironment, namespace, integrationTestScenario).
+		WithIntegrationLabels(integrationTestScenario).
+		WithSnapshot(snapshot).
+		AsEnvironment()
+
```

Comment:
We also need to make sure to set the `OwnerReference` for the ephemeral environment (like it's done e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L395)) so it will be deleted if the application is deleted.

Reply:
OwnerReference will be presented as follows:
```
ownerReferences:
    - apiVersion: [appstudio.redhat.com/v1alpha1](http://appstudio.redhat.com/v1alpha1)
      blockOwnerDeletion: true
      controller: true
      kind: Application
      name: example-application
      uid: 22de8273-cb3b-4888-bc66-8332d6a32c0a
```

Reply:
Ack, thanks!